### PR TITLE
Change the browser url,  PLATFORM-3904

### PR DESCRIPTION
**Pull request by abador**
Change the browser url for redirected pages on logged users

### CODE REVIEW

File: extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php [hunk 1]
```diff
@@ -357,4 +357,22 @@ public static function onLinkEnd( $skin, Title $target, array $options, &$text,
...
+
+	/**
+	 * Hook: set script that changes the address bar in the browser for the redirected url
+	 * @param Article $page
+	 * @return bool
+	 */
+	public static function onArticleViewRedirect( Article $page ): bool {
+		if( !$page->getContext()->getUser()->isAnon() ) {
+			$script = 'history.replaceState({}, "' . $page->getTitle()->getText().'", "' .
+					  $page->getTitle()->getLocalURL() . '");';
```
Comment (themech) [inline extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php]: After talking about it with @Grunny:

- this is not secure, one could create a Title that closes the quotes and injects a JS
- we should not inject inline JS this here - it is better to use JS file and already existing JS variables and trigger it from php; upstream code has it implemented: https://github.com/wikimedia/mediawiki/blob/master/resources/src/mediawiki.action/mediawiki.action.view.redirect.js & https://github.com/wikimedia/mediawiki/blob/master/includes/page/Article.php#L1093-L1098; can you please try to backport it?

**Review by themech — CHANGES_REQUESTED** [review]

**Review by harnash — APPROVED** [review]

File: extensions/wikia/SEOTweaks/SEOTweaks.setup.php [hunk 2]
```diff
@@ -23,5 +23,6 @@
 $wgHooks['OpenGraphMetaHeaders'][] = 'SEOTweaksHooksHelper::onOpenGraphMetaHeaders';
 $wgHooks['ShowMissingArticle'][] = 'SEOTweaksHooksHelper::onShowMissingArticle';
 
+
```
Comment (themech) [inline extensions/wikia/SEOTweaks/SEOTweaks.setup.php]:
```suggestion
```

File: extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php [hunk 3]
```diff
@@ -357,4 +357,6 @@ public static function onLinkEnd( $skin, Title $target, array $options, &$text,
 
 		return true;
 	}
+
```
Comment (themech) [inline extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php]:
```suggestion
```

File: extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php [hunk 4]
```diff
@@ -357,4 +357,6 @@ public static function onLinkEnd( $skin, Title $target, array $options, &$text,
...
+
```
Comment (themech) [inline extensions/wikia/SEOTweaks/SEOTweaksHooksHelper.class.php]:
```suggestion
```

File: includes/Article.php [hunk 5]
```diff
@@ -930,6 +930,18 @@ public function showRedirectedFromHeader() {
...
+
+				if( !$this->getContext()->getUser()->isAnon() ) {
+					// Add the script to update the displayed URL and
+					// set the fragment if one was specified in the redirect
+					$wgOut->addJsConfigVars( [
+						'wgInternalRedirectTargetUrl' => $this->getTitle()->getLocalURL(),
+					] );
+					$wgOut->addModules( [ 'mediawiki.action.view.redirect' ] );
+				}
+
```
Comment (themech) [inline includes/Article.php]:
```suggestion
```

**Review by themech — APPROVED** [review]

**Pull request merged**